src/sagemaker/lineage/lineage_trial_component.py

```diff
@@ -0,0 +1,184 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module contains code to create and manage SageMaker ``LineageTrialComponent``."""
+from __future__ import absolute_import
+
+import logging
+
+from typing import List
+
+from sagemaker.apiutils import _base_types
+from sagemaker.lineage.query import (
+    LineageQuery,
+    LineageFilter,
+    LineageSourceEnum,
+    LineageEntityEnum,
+    LineageQueryDirectionEnum,
+)
+from sagemaker.lineage.artifact import Artifact
+
+
+LOGGER = logging.getLogger("sagemaker")
+
+
+class LineageTrialComponent(_base_types.Record):
+    """An Amazon SageMaker, lineage trial component, which is part of a SageMaker lineage.
+
+    A trial component is a stage in a trial.
+    Trial components are created automatically within the SageMaker runtime and may not be
```

Comment:
Trial components can be created directly.

Reply:
> Trial components can be created directly.

updated